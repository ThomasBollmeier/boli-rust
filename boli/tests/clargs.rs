use assert_cmd::Command;

#[test]
fn test_command_line_args() {
    let mut cmd = Command::cargo_bin("boli").unwrap();

    let result = cmd
        .arg("tests/input/clargs.boli")
        .arg("arg1")
        .arg("-v2")
        .arg("--answer=42")
        .ok();

    assert!(result.is_ok());

    let output = String::from_utf8(result.unwrap().stdout).unwrap();
    let expected_output = std::fs::read_to_string("tests/output/clargs.out").unwrap();
    assert_eq!(output, expected_output);
}
