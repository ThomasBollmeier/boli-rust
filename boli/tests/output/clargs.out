3 command line arguments: 
arg1
-v2
--answer=42
nil
